# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: db39bf2058e1
# Full Hash: db39bf2058e13e89ac9238aa864df13a1c41a163
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-20 14:55:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 4: Untie frame iteration from Frame::tls. r=lth
#   
#   Here we replace usage of Frame::tls in frame iteration with GetNearestEffectiveTls.
#   We also maintain current tls for frame iteration object to not to call GetNearestEffectiveTls everytime.
#   
# ==============================================================================

diff -r a072d5ee2511 -r db39bf2058e1 js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Tue Oct 20 10:24:32 2020 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Tue Oct 20 10:24:40 2020 +0000
@@ -4618,19 +4618,22 @@
 template void AutoGenericRegisterScope<Register>::reacquire();
 template void AutoGenericRegisterScope<FloatRegister>::reacquire();
 
-wasm::TlsData* ExtractCallerTlsFromFrameWithTls(wasm::Frame* fp) {
-  return *reinterpret_cast<wasm::TlsData**>(
-      reinterpret_cast<uint8_t*>(fp) + sizeof(wasm::Frame) + ShadowStackSpace +
-      wasm::FrameWithTls::callerTLSOffset());
-}
-
-wasm::TlsData* ExtractCalleeTlsFromFrameWithTls(wasm::Frame* fp) {
-  return *reinterpret_cast<wasm::TlsData**>(
-      reinterpret_cast<uint8_t*>(fp) + sizeof(wasm::Frame) + ShadowStackSpace +
-      wasm::FrameWithTls::calleeTLSOffset());
-}
-
 #endif  // DEBUG
 
 }  // namespace jit
+
+namespace wasm {
+TlsData* ExtractCallerTlsFromFrameWithTls(Frame* fp) {
+  return *reinterpret_cast<TlsData**>(reinterpret_cast<uint8_t*>(fp) +
+                                      sizeof(Frame) + ShadowStackSpace +
+                                      FrameWithTls::callerTLSOffset());
+}
+
+TlsData* ExtractCalleeTlsFromFrameWithTls(Frame* fp) {
+  return *reinterpret_cast<TlsData**>(reinterpret_cast<uint8_t*>(fp) +
+                                      sizeof(Frame) + ShadowStackSpace +
+                                      FrameWithTls::calleeTLSOffset());
+}
+}  // namespace wasm
+
 }  // namespace js